### PR TITLE
ARROW-3692: [Gandiva][Ruby] Add Ruby bindings of Gandiva

### PR DESCRIPTION
**Pull request by shiro615**
It's Ruby bindings of Gandiva.

### CODE REVIEW

File: ruby/red-gandiva/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Red Gandiva uses [Gandiva GLib](https://github.com/apache/arrow/tree/master/c_glib/gandiva-glib) and [gobject-introspection gem](https://rubygems.org/gems/gobject-introspection) to generate Ruby bindings of Gandiva.
+
+Gandiva GLib is a C wrapper for [Gandiva C++](https://github.com/apache/arrow/tree/master/cpp/gandiva). GObject Introspection can't use Gandiva C++ directly. Gandiva GLib is a bridge between Gandiva C++ and GObject Introspection.
+
+gobject-introspection gem is a Ruby bindings of GObject Introspection. Red Gandiva uses GObject Introspection via gobject-introspection gem.
+
+## Install
+
+Install Gandiva GLib before install Red Gandiva. Use [packages.red-data-tools.org](https://github.com/red-data-tools/packages.red-data-tools.org) for installing Gandiva GLib.
+
```
Comment (kou) [inline ruby/red-gandiva/README.md]: We didn't create packages for Gandiva and Gandiva GLib but it's OK. Because we will create them until 0.12.0. :-)

**Review by kou — APPROVED** [review]
Thanks!
I'll merge this when CI is green.

File: ruby/red-gandiva/README.md (hunk 1 above)
Comment (shiro615) [reply ruby/red-gandiva/README.md]: OK! Thanks!

**Comment by shiro615** [comment]
> I'll merge this when CI is green.

I'll try to fix it.

**Comment by kou** [comment]
We need at least one test for `ruby/red-gandiva/test/run-test.rb` for CI failure on Linux.
I've added it.

**Comment by kou** [comment]
I noticed that we need to add some convenient methods to `Arrow::Schema` to write `schema[:field1]`.
Can you work on it in a follow-up pull request?

**Comment by kou** [comment]
The added test doesn't need but we should add `schema.each {|field| ...}` API.

**Comment by shiro615** [comment]
> We need at least one test for ruby/red-gandiva/test/run-test.rb for CI failure on Linux.
> I've added it.

I see. Thanks!

> I noticed that we need to add some convenient methods to Arrow::Schema to write schema[:field1].
> Can you work on it in a follow-up pull request?

> The added test doesn't need but we should add schema.each {|field| ...} API.

Yes! I'll send follow-up pull requests.

**Comment by kou** [comment]
Linux CI is green.
macOS CI will be passed with #2897.

> Yes! I'll send follow-up pull requests.

Thanks! You can work on it before this pull request is merged. If these follow-up pull requests are merged before this pull request is merged, we can use the features introduced by these follow-up pull requests in this pull request.

**Comment by shiro615** [comment]
Thanks!
I'll work it before this pull request is merged.

**Comment by shiro615** [comment]
I've created a pull request: #2900.

**Comment by kou** [comment]
I'll rebase on master to use #2897 and #2900.

**Comment by kou** [comment]
CI is green.
I'll merge this.

**Comment by shiro615** [comment]
Thanks!

**Pull request closed without merging**